php/libraries/VisitController.class.inc

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * This file contains the VisitController class.
+ *
+ * PHP Version 7
+ *
+ * @category Controller
+ * @package  Visits
+ * @author   Mélanie Legault <melanie.legault2@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS;
+
+/**
+ * Base class that contains method to access the visit of projects
+ *
+ * @category Controller
+ * @package  Visits
+ * @author   Mélanie Legault <melanie.legault2@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class VisitController
+{
+    /**
+     * The database connection to use
+     *
+     * @var \Database
+     */
+    protected $database;
+
+    /**
+     * The constructor
+     *
+     * @param \Database $database The database connection to use
+     */
+    public function __construct(\Database $database)
+    {
+        if (!$database->isConnected()) {
+            throw new \RuntimeException("VisitController - Database not connected");
+        }
+        $this->database = $database;
+    }
+
+    /**
+     * Retruns a list of configured Visits
+     *
+     * @return array The list of visits
+     */
+    public function getAllVisits(): array
+    {
+        return array_map(
+            function ($row) {
+                return new \LORIS\Visit(
+                    $row['visit']
+                );
+            },
+            $this->database->pselect(
+                'SELECT
+                 v.VisitName as "visit"
+                FROM
+                 visit v
+                ',
+                array()
+            )
+        );
+    }
+
+    /**
+     * Retruns a VisitID by name
+     *
+     * @param string $visit the name of the visit
+     *
+     * @return array the ID of the visit
+     */
+    public function getVisitsWithID(string $visit): array
+    {
+        return array_map(
+            function ($row) {
+                return new \LORIS\Visit(
+                    $row['ID']
+                );
+            },
+            $this->database->pselect(
+                'SELECT
+                 v.VisitID as "ID"
+                FROM
+                 visit v
+                WHERE
+                 v.VisitName = :name
+                ',
+                array('name' => $visit)
+            )
+        );
+    }
+    /**
```

Comment:
sismilar issues with the above function, first it reads "withProject" while project is never really supplied ? this function to me should be more like `getVisits` or maybe `getVisitsAssociation` or `getVisitsForAllProjects`

when i see **with** i'm expecting something like 
`$newVisit` = $oldVisit->withName("Rida")` where rida will be set as the $oldVisit's name and returned into a $newVisit object